### ISSUE TITLE
What if user id is not integer?

### ISSUE THREAD

**Issue by danaki** [yokai-php/messenger-bundle#29]
In my project, I defined user ids as uuid. When I use the built-in 'doctrine' channel, I get the following error:
```
"An error has occurred during the notification process of message \"transaction_state_change\" for channel \"Yokai\\MessengerBundle\\Channel\\DoctrineChannel\". Exception : Doctrine\\DBAL\\DBALException - An exception occurred while executing 'INSERT INTO yokai_messenger_notification (subject, body, recorded_at, recipient_class, recipient_id, delivered_at) VALUES (?, ?, ?, ?, ?, ?)' with params [\"Transaction status change\", \"0c0bbf47-f5ba-4b3c-9aa7-8810ca97230c\\n\", \"2018-09-04 20:24:58\", \"App\\\\Entity\\\\User\", \"a1385654-fb9e-48ef-864b-9c9de4b8193f\", null]:\n\nNotice: Object of class App\\Entity\\Id\\UserUuid could not be converted to int"
```

 * How can I override the Notification entity to use uuid as the user type? (May be [this issue](https://github.com/doctrine/doctrine2/issues/6049) can help?
 * I'm using your yokai-token-bundle and there [you define](https://github.com/yokai-php/security-token-bundle/blob/master/Resources/config/doctrine/Token.orm.xml) user id as varchar(255) which is ok in my case.
 * If the only way to override bundle mappings is thru overriding an xml config, it would be great to reflect it in the documentation because I guess it's an often used case when people want to bind Notification entity to User thru relations and vice versa.

**Comment by danaki** [comment]
Seems that overriding xml mappings doesn't work, here another issue I've just opened https://github.com/doctrine/DoctrineBundle/issues/846

**Comment by yann-eugone** [comment]
The easiest solution for me is to change the field type to a string.

Mapping overriding in doctrine is painful, I don't want to document something like this.

The last possible solution I see, is that you implement your own channel.

**Comment by danaki** [comment]
Yes, convert id to string just for consistency with your other bundle. You see it appears that overriding in doctrine is broken or not documented enough, let's see how the related issue resolves. I've checked the source, there's nothing wrong in built-in doctrine channel, it's type-agnostic against recpientId.

**Comment by danaki** [comment]
May be it's even better if messenger-bundle doesn't provide Entity at all but requires it's user to create such Notification entity (of course with known interface) by himself and configure the bundle providing the class name then?

**Comment by yann-eugone** [comment]
Yes, it is. But currenly I've not enough time to prepare the bundle for this.

(I'm already thinking about the next major, with more abstraction everywhere)
